sklearn/neighbors/classification.py

```diff
@@ -388,54 +444,115 @@ def predict(self, X):
         -------
         y : array of shape [n_samples] or [n_samples, n_outputs]
             Class labels for each data sample.
+        """
+
+        probs = self.predict_proba(X)
+        classes_ = self.classes_
+
+        if not self.outputs_2d_:
+            probs = [probs]
+            classes_ = [self.classes_]
 
+        n_outputs = len(classes_)
+        n_samples = probs[0].shape[0]
+        y_pred = np.empty((n_samples, n_outputs),
+                          dtype=classes_[0].dtype)
+
+        for k, prob in enumerate(probs):
```

Comment:
Maybe add a small comment here again saying that it iterates over outputs.

Reply:
Added